samtranslator/plugins/application/serverless_app_plugin.py

```diff
@@ -372,9 +399,6 @@ def _sar_service_call(self, service_call_lambda, logical_id, *args):
             error_code = e.response["Error"]["Code"]
             if error_code in ("AccessDeniedException", "NotFoundException"):
                 raise InvalidResourceException(logical_id, e.response["Error"]["Message"])
-
```

Comment:
Is there any reason we shouldn't do the Throttle handling within this (`_sar_service_call`) function? Seems odd to do the checking in multiple places.

Reply:
The check for throttling and the retry is only in one place (where we're trying to retrieve the CloudFormation template).  The SAR Create call currently does not have a retry and we have not yet seen it get throttled.  If we start seeing it, we could implement a retry in `_sar_service_call`.